### PR TITLE
doc: godoc and examples for outofband client pkg

### LINKED ISSUE #1789: Out-of-band documentation
Add usage, api, godoc, etc documentation

https://github.com/hyperledger/aries-framework-go/issues/1785#issuecomment-624658017 - @llorllale

### PR DESCRIPTION
**Pull request by llorllale**
closes #1789

Signed-off-by: George Aristy <george.aristy@securekey.com>

### CODE REVIEW

File: pkg/client/outofband/doc.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+// }
+//
+// You can create requests and invitations with client.CreateRequest() and client.CreateInvitation()
+// respectively.
+//
+// You can accept out-of-band requests and invitations received via out of band channels.
+// If you have a request or an invitation on hand you can use the client.AcceptRequest() and
+// client.AcceptInvitation() respectively. These return the ID of the newly-created connection
+// record.
+//
```
Comment (troyronda) [inline pkg/client/outofband/doc.go:25]: Do you have a code example of this scenario?

Comment (rolsonquadras) [reply pkg/client/outofband/doc.go:25]: https://github.com/hyperledger/aries-framework-go/blob/master/pkg/client/didexchange/example_test.go

Comment (llorllale) [reply pkg/client/outofband/doc.go:25]: I'm working on adding example tests as per @rolsonquadras 's suggestion.

Comment (llorllale) [reply pkg/client/outofband/doc.go:25]: But to answer your question: yes, in the out-of-band BDD tests, example: https://github.com/hyperledger/aries-framework-go/blob/master/test/bdd/pkg/outofband/outofband_sdk_steps.go#L158

Comment (llorllale) [reply pkg/client/outofband/doc.go:25]: @rolsonquadras @troyronda please check now

**Pull request merged**